File: LLM-Evaluation/src/main/org/cornell/TaskManager.java
package cornell;

import java.util.ArrayList;
import java.util.List;

public class TaskManager {
    private final List<Task> tasks = new ArrayList<>();

    public void addTask(Task task) {
        tasks.add(task);
    }

    public void removeTask(Task task) {
        tasks.remove(task);
    }

    public void markTaskAsCompleted(Task task) {
        if (tasks.contains(task)) {
            task.markAsCompleted();
        }
    }

    public List<Task> getAllTasks() {
        return new ArrayList<>(tasks);
    }

    public List<Task> getPendingTasks() {
        List<Task> pendingTasks = new ArrayList<>();
        for (Task task : tasks) {
            if (!task.isCompleted()) {
                pendingTasks.add(task);
            }
        }
        return pendingTasks;
    }

    public List<Task> getCompletedTasks() {
        List<Task> completedTasks = new ArrayList<>();
        for (Task task : tasks) {
            if (task.isCompleted()) {
                completedTasks.add(task);
            }
        }
        return completedTasks;
    }

    public void shiftTaskAround(Task task, int position) {
        if (position >= tasks.size() || !tasks.contains(task)){
            return;
        }
        tasks.remove(task);
        tasks.add(position, task);
    }

    public List<Task> getTasksOwnedBy(String owner) {
        List<Task> rlTask = new ArrayList<>();
        for (Task t: tasks) {
            if (t.getOwner().equals(owner)) {
                rlTask.add(t);
            }
        }
        return rlTask;
    }

    public void changeOwner(String owner, Task task) {
        if (tasks.contains(task)) {
            int i = tasks.indexOf(task);
            tasks.get(i).setOwner(owner);
        }
    }
}

File: translation/2024-11-21_13-56-51/run3/TaskManager.py
import Task

class TaskManager:
    def __init__(self):
        self.tasks = []

    def addTask(self, task):
        self.tasks.append(task)

    def removeTask(self, task):
        self.tasks.remove(task)

    def markTaskAsCompleted(self, task):
        if task in self.tasks:
            task.markAsCompleted()

    def getAllTasks(self):
        return list(self.tasks)

    def getPendingTasks(self):
        pending_tasks = []
        for task in self.tasks:
            if not task.isCompleted():
                pending_tasks.append(task)
        return pending_tasks

    def getCompletedTasks(self):
        completed_tasks = []
        for task in self.tasks:
            if task.isCompleted():
                completed_tasks.append(task)
        return completed_tasks

    def shiftTaskAround(self, task, position):
        if position >= len(self.tasks) or task not in self.tasks:
            return
        self.tasks.remove(task)
        self.tasks.insert(position, task)

    def getTasksOwnedBy(self, owner):
        rl_task = []
        for t in self.tasks:
            if t.getOwner() == owner:
                rl_task.append(t)
        return rl_task

    def changeOwner(self, owner, task):
        if task in self.tasks:
            i = self.tasks.index(task)
            task.setOwner(owner)
